### ISSUE TITLE
Enriching text on multi value strings erases them

### ISSUE THREAD

**Issue by DavidLeoni** [opendatatrentino/OpenDataRise#152]
In step 4 enriching text on multi value semantic text sets cells to null. Works fine when there is only one item in the array

**Comment by DavidLeoni** [comment]
Nastier to solve than I thought, setting it to P2

**Comment by DavidLeoni** [comment]
@gbella keep in mind this also affects cells holding lists of entity names. If entities are in long text (i.e. Prodotto nella Valle dell'Adige nei comuni di Andalo, Amblar, Bresimo, ... ), it's not an issue, but if they are in a form ODR thinks it could be an easy to parse list such as 'Andalo', Amblar, 'Bresimo' then user will be asked to create an array in step 3 and in step 4 enrichment will fail.

**Comment by DavidLeoni** [comment]
Resetting it to P1 - better to deal with it once and for all

Related to #153

**Comment by DavidLeoni** [comment]
Should be fixed in dev-0.3

**Comment by DavidLeoni** [comment]
Reopening, found this exception when enrichment fails (`if (entitiesResults == null) ...`)

```

03:03:2016:20:04:27.299 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Preparing EnrichProcess... 
03:03:2016:20:04:27.299 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - EnrichProcess has started.
03:03:2016:20:04:27.299 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 0
03:03:2016:20:04:27.301 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 1
03:03:2016:20:04:27.302 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 2
03:03:2016:20:04:27.302 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 3
03:03:2016:20:04:27.303 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 4
03:03:2016:20:04:27.303 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 5
03:03:2016:20:04:27.304 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 6
03:03:2016:20:04:27.304 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Collecting cell at row 7
03:03:2016:20:04:27.305 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Number of rows with missing values that will be skipped: 0
03:03:2016:20:04:27.305 [Thread-24] INFO  e.t.o.o.model.EnrichProcess - Enriching with reconciliation 8 entities...
Exception in thread "Thread-24" java.lang.ClassCastException: java.util.ArrayList cannot be cast to eu.trentorise.opendata.semantics.model.entity.IEntity
        at eu.trentorise.opendata.opendatarise.model.EnrichProcess.enrichWithReconciliation(EnrichProcess.java:669)
        at eu.trentorise.opendata.opendatarise.model.EnrichProcess.run(EnrichProcess.java:182)
        at java.lang.Thread.run(Thread.java:745)
03:03:2016:20:04:27.318 [qtp9272003-28] INFO  refine - request.getPathInfo() = /command/core/get-history
03:03:2016:20:04:27.318 [qtp9272003-28] INFO  refine - GET /command/core/get-history
```

**Comment by DavidLeoni** [comment]
Fixed the bug in dev-0.3, maybe it was one of the causes of #182 as well
